sa/storage-authority.go

```diff
@@ -237,6 +237,46 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64
 	return ssa.GetAuthorization(auth.ID)
 }
 
+type TooManyCertificatesError string
+
+func (t TooManyCertificatesError) Error() string {
+	return string(t)
+}
+
+// CountCertificates returns the number of certificates issued within a time
+// period containing DNSNames that are equal to, or subdomains of, the given name.
+func (ssa *SQLStorageAuthority) CountCertificatesByName(domain string, earliest, latest time.Time) (int, error) {
+	var count int64
+	const max = 10000
+	var serials []struct {
+		Serial string
+	}
+	_, err := ssa.dbMap.Select(
+		&serials,
+		`SELECT serial from issuedNames
+		 WHERE (reversedName = :reversedDomain OR
+			      reversedName LIKE CONCAT(:reversedDomain, ".%"))
+		 AND Issued > :earliest AND Issued <= :latest
+		 LIMIT :limit;`,
+		map[string]interface{}{
+			"reversedDomain": core.ReverseName(domain),
+			"earliest":       earliest,
+			"latest":         latest,
+			"limit":          max + 1,
+		})
+	if err != nil {
+		return -1, err
+	} else if count > max {
+		return -1, TooManyCertificatesError(fmt.Sprintf("More than %d issuedName entries for %s.", max, domain))
+	}
+	serialMap := make(map[string]struct{}, len(serials))
+	for _, s := range serials {
+		serialMap[s.Serial] = struct{}{}
+	}
+
+	return len(serialMap), nil
```

Comment:
Since you only return the count, it seems like this should be a `SELECT COUNT(serial)...` query.